File: apps/api/app/schemas/notification.py
import uuid
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict
from app.models.enums import NotificationType


class NotificationResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    type: NotificationType
    title: str
    body: str
    requisition_id: Optional[uuid.UUID] = None
    application_id: Optional[uuid.UUID] = None
    read_at: Optional[datetime] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class NotificationListResponse(BaseModel):
    items: List[NotificationResponse]
    unread_count: int


class NotificationReadRequest(BaseModel):
    notification_ids: Optional[List[uuid.UUID]] = None
    mark_all: bool = False


class NotificationReadResponse(BaseModel):
    marked_count: int
    unread_count: int

File: apps/web/src/types/notification.ts
export interface NotificationItem {
  id: string;
  user_id: string;
  type: string;
  title: string;
  body: string;
  requisition_id?: string | null;
  application_id?: string | null;
  read_at?: string | null;
  created_at: string;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  unread_count: number;
}
